### PR TITLE
pkg/trace/obfuscate: fix SQL obfuscation errors on non-alphanumeric literals

### PR DESCRIPTION
**Pull request by alexandre-normand**
### What does this PR do?

This fixes obfuscation errors on valid queries like ``select count(*) as `count(*)` from test``. 

Previously resulting in obfuscation errors like:
```
at position 25: literal identifiers must end in "`", got "(" (40)
```

### Additional Notes

Instead of being strict about what characters are allowed inside ``  ` `` ``  ` ``, this uses the scanString 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/trace/obfuscate/sql_test.go [hunk 1]
```diff
@@ -545,6 +545,21 @@ func TestSQLTableFinderAndReplaceDigits(t *testing.T) {
 				tables:     "test",
 				obfuscated: "select * from test where ! - ?",
 			},
+			{
+				query:      "select count(*) as `count(*)` from test",
```
Comment (gbbr) [inline pkg/trace/obfuscate/sql_test.go:549]: Can you please remove the keys from these structs? Including the ones you've added in your last PR. Just to stay consistent. Or, if you feel strongly about having keys, then please add them everywhere in the slice.

**Review by gbbr — APPROVED** [review]
👍🏻 Glad to see we're getting improvements.

File: pkg/trace/obfuscate/sql_test.go (hunk 1 above)
Comment (alexandre-normand) [reply pkg/trace/obfuscate/sql_test.go]: I do prefer when keys are defined but I actually didn't do it on purpose here (I was using IntelliJ's "fill in all keys" which did this) and this isn't a case I feel strongly about. 

Updated ✅

**Comment by alexandre-normand** [comment]
> 👍🏻 Glad to see we're getting improvements.

😁

**Pull request merged**